### PR TITLE
Update Config Setup

### PR DESCRIPTION
**Pull request by ltcarbonell**
Pulls out the intermediary struct that is used for config mapping

### CODE REVIEW

File: benchmarktests/target_auth_approle.go [hunk 1]
```diff
@@ -85,7 +80,9 @@ type SecretIDConfig struct {
 // test configuration in Vault. Any default configuration definitions for required
 // parameters will be set here.
 func (a *ApproleAuth) ParseConfig(body hcl.Body) error {
-	a.config = &ApproleTestConfig{
+	testConfig := &struct {
+		Config *ApproleAuthTestConfig `hcl:"config,block"`
+	}{
```
Comment (elliesterner) [inline benchmarktests/target_auth_approle.go:85]: I get why were doing this...The top level struct does seem a bit redundant. But I think this is more confusing :/ ... I could be just used to the other way, and will defer to what others think!

Comment (davidadeleon) [reply benchmarktests/target_auth_approle.go:85]: The top level struct only really exists to allow for the decoding to work as expected here, and we're generally only concerned with the inner struct anyway. While it is more verbose, having this outer struct made naming a bit more difficult considering we really didn't do anything with it after handling the HCL decoding.

**Review by davidadeleon — APPROVED** [review]
Looks good! One small thing we may want to think about, but nothing that prevents this from being merged.

File: benchmarktests/target_auth_cert.go [hunk 2]
```diff
@@ -108,7 +106,7 @@ func (c *CertAuth) GetTargetInfo() TargetInfo {
 func (c *CertAuth) Setup(client *api.Client, randomMountName bool, mountName string) (BenchmarkBuilder, error) {
 	var err error
 	authPath := mountName
-	config := c.config.Config
+	config := c.config
```
[the remark this answers is not in the document]
Comment (ltcarbonell) [reply benchmarktests/target_auth_cert.go]: Good note.  I cleaned those up

**Pull request merged**